widget/fs.lua

```diff
@@ -131,6 +131,10 @@ local function factory(args)
         fs.notification_preset.text = tconcat(notifytable)
     end
 
+    function fs.update()
+        Gio.Async.start(update_synced)()
```

Comment:
I would suggest to use `Gio.Async.start(gears.protected_call)(update_synced)`.
Try adding `error("shit happens")` somewhere in `update_synced`. Before this commit, AwesomeWM should complain about an error in a callback and continue along. After this commit, Lua will abort the process due to an unprotected error. With my change, `gears.protected_call` we are back at an error message being printed and things continue to chuck along.

Reply:
Good to know, I'll fix it.
I don't know how well it works to add commits to an already merged pull request. Should I just open a new one?

Reply:
> Should I just open a new one?

Yes.

Reply:
> Shouldn't it be gears.protected_call.call, or did I get something wrong?

It does not really matter since both code paths do the same thing:
https://github.com/awesomeWM/awesome/blob/538586c170cffebca405c005b63f68b125f3d340/lib/gears/protected_call.lua#L47-L56

Reply:
> It does not really matter since both code paths do the same thing

@psychon Well, I'm getting an error when using just `gears.protected_call` with `Gio.Async.start`.
It's probably doing some type checking, without considering it being callable.
```lua
local gears = require("gears")
Gio.Async.start(gears.protected_call)(update_synced)
```
```
error while running function!
stack traceback:
	[C]: in function 'coroutine.create'
	/usr/share/lua/5.3/lgi/override/Gio.lua:57: in field 'start'
	/home/marvin/.config/awesome/lain/widget/fs.lua:138: in local 'func'
	/usr/share/awesome/lib/gears/object.lua:148: in function 'gears.object.emit_signal'
	/home/marvin/.config/awesome/lain/helpers.lua:97: in function 'lain.helpers.newtimer'
	/home/marvin/.config/awesome/lain/widget/fs.lua:151: in function 'lain.widget.fs'
	/home/marvin/.config/awesome/widgets/fs.lua:69: in function 'widgets.fs.create'
	/home/marvin/.config/awesome/widgets/base.lua:12: in function 'widgets.fs.allocate'
	(...tail calls...)
	/home/marvin/.config/awesome/rc.lua:243: in main chunk
error: /usr/share/lua/5.3/lgi/override/Gio.lua:57: bad argument #1 to 'create' (function expected, got table)
2020-11-29 14:19:01 E: awesome: Error during a protected call: /home/marvin/.config/awesome/lain/widget/fs.lua:141: attempt to call a table value (upvalue 'callback')
```

Reply:
> It's probably doing some type checking, without considering it being callable.

Ah, sorry. I did not know.